File: src/main/java/eu/fayder/restcountries/v2/domain/Country.java
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
package eu.fayder.restcountries.v2.domain;

import eu.fayder.restcountries.domain.BaseCountry;

import java.util.List;

public class Country extends BaseCountry {

    private List<Currency> currencies;
    private List<Language> languages;
    private Translations translations;
    private String flag;
    private List<RegionalBloc> regionalBlocs;
    private String cioc;

    public List<Currency> getCurrencies() {
        return currencies;
    }

    public List<Language> getLanguages() {
        return languages;
    }

    public Translations getTranslations() {
        return translations;
    }

    public String getFlag() {
        return flag;
    }

    public List<RegionalBloc> getRegionalBlocs() {
        return regionalBlocs;
    }

    public String getCioc() {
        return cioc;
    }
}

File: src/main/kotlin/eu/fayder/restcountries/v3/domain/Country.kt
package eu.fayder.restcountries.v3.domain

import eu.fayder.restcountries.domain.BaseCountry

data class Country
(
        val currencies: List<Currency>?,
        val languages: List<Language>,
        val flag: String,

        // From mledoze
        val names: Names,
        val translations: Translations,
        val cioc: String
) : BaseCountry()